Sources/NIOPosix/PosixSingletons+ConcurrencyTakeOver.swift

```diff
@@ -0,0 +1,123 @@
+            fatalError("Must be called only once")
+        }
+
+        #if canImport(Darwin)
+        guard pthread_main_np() == 1 else {
+            fatalError("Must be called from the main thread")
+        }
+        #endif
+
+        // Unsafe 1: We pretend that the hook's type is actually fully equivalent to `ConcurrencyEnqueueGlobalHook`
+        //   @convention(thin) (UnownedJob, @convention(thin) (UnownedJob) -> Void) -> Void
+        // which isn't formally guaranteed.
+        let concurrencyEnqueueGlobalHookPtr = dlsym(
+            dlopen(nil, RTLD_NOW),
+            "swift_task_enqueueGlobal_hook"
+        )?.assumingMemoryBound(to: UnsafeRawPointer?.AtomicRepresentation.self)
+        guard let concurrencyEnqueueGlobalHookPtr = concurrencyEnqueueGlobalHookPtr else {
+            return false
+        }
+
+        // We will use an atomic operation to swap the pointers aiming to protect against other code that attempts
+        // to swap the pointer. This isn't guaranteed to work as we can't be sure that the other code will actually
+        // use atomic compares and exchanges to. Nevertheless, we're doing our best.
+        let concurrencyEnqueueGlobalHookAtomic = UnsafeAtomic<UnsafeRawPointer?>(at: concurrencyEnqueueGlobalHookPtr)
+        // note: We don't need to destroy this atomic as we're borrowing the storage from `dlsym`.
+
+        return withUnsafeTemporaryAllocation(
+            of: ConcurrencyEnqueueGlobalHook.self,
+            capacity: 1
+        ) { enqueueOnNIOPtr -> Bool in
+            // Unsafe 2: We mandate that we're actually getting _the_ function pointer to the closure below which
+            // isn't formally guaranteed by Swift.
+            enqueueOnNIOPtr.baseAddress!.initialize(to: { job, _ in
+                // This formally picks a random EventLoop from the singleton group. However, `EventLoopGroup.any()`
+                // attempts to be sticky. So if we're already in an `EventLoop` that's part of the singleton
+                // `EventLoopGroup`, we'll get that one and be very fast (avoid a thread switch).
+                let targetEL = MultiThreadedEventLoopGroup.singleton.any()
+
+                (targetEL.executor as! any SilenceWarning).enqueue(job)
+            })
+
+            // Unsafe 3: We mandate that the function pointer can be reinterpreted as `UnsafeRawPointer` which isn't
+            // formally guaranteed by Swift
+            return enqueueOnNIOPtr.baseAddress!.withMemoryRebound(
+                to: UnsafeRawPointer.self,
+                capacity: 1
+            ) { enqueueOnNIOPtr in
+                // Unsafe 4: We just pretend that we're the only ones in the world pulling this trick (or at least
+                // that the others also use a `compareExchange`)...
+                guard concurrencyEnqueueGlobalHookAtomic.compareExchange(
+                    expected: nil,
+                    desired: enqueueOnNIOPtr.pointee,
+                    ordering: .relaxed
```

Comment:
Similarly this should probably be a `releasing` ordering. Again, `.enqueue` should be an appropriate compiler barrier so arguably unnecessary, but I'd rather we did this right.